src/Symfony/Component/Cache/Traits/RedisTrait.php

```diff
@@ -167,7 +180,14 @@ public static function createConnection($dsn, array $options = array())
             $params['scheme'] = $scheme;
             $params['database'] = $params['dbindex'] ?: null;
             $params['password'] = $auth;
-            $redis = new $class((new Factory())->create($params));
+
+            $predisOptions = $params['predis_options'] ?? array();
+
+            if (!\is_array($predisOptions)) {
```

Comment:
We are not doing this kind of checks anywhere in the code base, not sure if we want it here.

Reply:
What would you recommend instead? If we don't check and throw an exception predis will, I would imagine this would suffice?

Reply:
Letting predis handle the situation seems better.